### DISCUSSION TITLE
2024-06-11-cmake-%E8%AE%BE%E7%BD%AE%E9%BB%98%E8%AE%A4%E7%BC%96%E8%AF%91%E5%99%A8cmake_cxx_compiler%E9%80%A0%E6%88%90%E6%AD%BB%E5%BE%AA%E7%8E%AF%E7%9A%84%E9%97%AE%E9%A2%98/

### DISCUSSION THREAD

**Discussion by giscus** [williamlfang/williamlfang.github.io#6] [Announcements]
# 2024-06-11-cmake-%E8%AE%BE%E7%BD%AE%E9%BB%98%E8%AE%A4%E7%BC%96%E8%AF%91%E5%99%A8cmake_cxx_compiler%E9%80%A0%E6%88%90%E6%AD%BB%E5%BE%AA%E7%8E%AF%E7%9A%84%E9%97%AE%E9%A2%98/

为了以指定版本的编译器进行编译，我们可以通过修改 cmake 默认编译器 (CMAKE_CXX_COMPILER)。然而，如果在多项目的 add_subdirectory 包含其他项目，可能造成死循环的问题。

https://williamlfang.github.io/2024-06-11-cmake-%E8%AE%BE%E7%BD%AE%E9%BB%98%E8%AE%A4%E7%BC%96%E8%AF%91%E5%99%A8cmake_cxx_compiler%E9%80%A0%E6%88%90%E6%AD%BB%E5%BE%AA%E7%8E%AF%E7%9A%84%E9%97%AE%E9%A2%98/

**Comment by williamlfang** [comment]
这个问题看起来比较 tricky，因为如果一开始没有设置 CMAKE_CXX_COMPILER，project 会重新执行环境变量，就一直会检查到 add_subdirectory 下面的 project 版本不一致了
